printrun/gcview.py

```diff
@@ -150,9 +150,14 @@ def draw_objects(self):
 
         for obj in self.parent.objects:
             if not obj.model \
-               or not obj.model.loaded \
-               or not obj.model.initialized:
+               or not obj.model.loaded:
                 continue
+            # Skip (comment out) initialized check, which safely causes empty 
+            # model during progressive load. This can cause exceptions/garbage
+            # render, but seems fine for now
+            # May need to lock init() and draw_objects() together
+            # if not obj.model.initialized:
+            #     continue
```

Comment:
This should be fine - it should never cause an error, but may cause an out-of-date object to be rendered during translation/rotation. I don't think it's a concern for us.

Reply:
Not looking at the code now, but can't imagine how interaction through mouse/keyboard events and rendering can mix in the main thread. The sequence should be:
mouse_event_handler() {rotation = ...; Refresh()};
draw() { model.init(); model.draw()}

Reply:
In that case it's not a concern at all - if the draw never happens before an init then we'll never see an out of date render either.